org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/LinkedResourcesUpdater.java

```diff
@@ -73,6 +81,14 @@ private void removeOldLinkedResources(StringSetProjectProperty knownLinkedResour
         }
     }
 
+    private Collection<String> getKnownLinkedResources(IProject project) throws CoreException {
+        String serializedForm = project.getPersistentProperty(LINKED_RESOURCES_KEY);
+        if (serializedForm == null) {
+            return Collections.emptyList();
+        }
+        return Splitter.on(File.pathSeparator).omitEmptyStrings().splitToList(serializedForm);
```

Comment:
Yes good idea, though probably some utility we delegate to instead of a superclass.